src/diffusers/models/embeddings.py

```diff
@@ -532,7 +617,10 @@ def apply_rotary_emb(
         else:
             raise ValueError(f"`use_real_unbind_dim={use_real_unbind_dim}` but should be -1 or -2.")
 
-        out = (x.float() * cos + x_rotated.float() * sin).to(x.dtype)
+        if upcast:
```

Comment:
ok but I'm wondering why upcasting will cause not able to generate videos for cogvideox
maybe we can test out not downcast rotary embedding instead (all our other pipelines keep it in float32)

Reply:
I mainly added this to numerically match the output of apply_rotary_emb with original implementation. It seems that even if we compute in float32, the results are great and the small diff does not impact quality, now that other bugs have been resolved. We can undo this change